app/src/main/java/cz/muni/fi/pv256/movio2/uco_433419/FilmAdapter.kt

```diff
@@ -2,23 +2,59 @@ package cz.muni.fi.pv256.movio2.uco_433419
 
 import android.support.v7.widget.RecyclerView
 import android.view.LayoutInflater
+import android.view.View
 import android.view.ViewGroup
 import android.widget.TextView
 
-/**
- * Created by Tomas on 20. 10. 2017.
- */
-class FilmAdapter(private var dataSet: ArrayList<Film>)  : RecyclerView.Adapter<FilmViewHolder>() {
+
+class FilmAdapter(private val dataSet: ArrayList<ListItem>, private val fragment: ListFragment)  : RecyclerView.Adapter<FilmAdapter.FilmViewHolder>() {
+    private val TYPE_FILM = 0
+    private val TYPE_CATEGORY = 1
 
     override fun onBindViewHolder(holder: FilmViewHolder?, position: Int) {
-        holder?.filmView?.text = dataSet[position].title
+        val viewType = getItemViewType(position)
+        if (viewType == TYPE_CATEGORY){
+            (holder?.view as TextView).text = dataSet[position].title
+        }
+        else {
+            (holder?.view as FilmView).setFilmProperties(dataSet[position] as Film)
+        }
+
     }
 
     override fun getItemCount(): Int = dataSet.count()
 
     override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): FilmViewHolder {
-        val v = LayoutInflater.from(parent?.context)
-                .inflate(R.layout.film_item, parent, false) as TextView
-        return FilmViewHolder(v)
+        return if (viewType == TYPE_CATEGORY){
+            val v = LayoutInflater.from(parent?.context)
+                    .inflate(R.layout.category_item, parent, false) as TextView
+            FilmViewHolder(v)
+        }
+        else {
+            val v = LayoutInflater.from(parent?.context)
+                    .inflate(R.layout.film_item, parent, false) as FilmView
+            FilmViewHolder(v)
+        }
+
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return if (dataSet[position] is Film)
```

Comment:
food for thought:

kedze Film a Category su dva rozdielne objekty, mohli by uz implicitne obsahovat svoj typ TYPE_FILM, TYPE_CATEGORY.
tato metoda by sa dala zjednodusit na return dataSet[position].getType().

Reply:
opravene:
9c0398b363bbcb615b0e074179d6568c708e996a